README.md

```diff
@@ -80,6 +81,103 @@ kt_jvm_library(
 )
 ```
 
+## Custom toolchain
+
+To enable a custom toolchain (to configure language level, etc.)
+do the following.  In a `<workspace>/BUILD.bazel` file define the following:
+
+```
+load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "define_kt_toolchain")
+
+define_kt_toolchain(
+    name = "kotlin_toolchain",
+    api_version = KOTLIN_LANGUAGE_LEVEL,  # "1.1", "1.2", or "1.3"
+    jvm_target = JAVA_LANGUAGE_LEVEL, # "1.6" or "1.8"
+    language_version = KOTLIN_LANGUAGE_LEVEL,  # "1.1", "1.2", or "1.3"
+)
+```
+
+and then in your `WORKSPACE` file, instead of `kt_register_toolchains()` do
+
+```
+register_toolchains("//:kotlin_toolchain")
+```
+
+## Custom kotlinc distribution (and version)
+
+To choose a different kotlinc distribution (only 1.3 variants supported), do the following
+in your `WORKSPACE` file (or import from a `.bzl` file:
+
+```
+load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kotlin_repositories")
+
+KOTLIN_VERSION = "1.3.31"
+KOTLINC_RELEASE_SHA = "107325d56315af4f59ff28db6837d03c2660088e3efeb7d4e41f3e01bb848d6a"
+
+KOTLINC_RELEASE = {
+    "urls": [
+        "https://github.com/JetBrains/kotlin/releases/download/v{v}/kotlin-compiler-{v}.zip".format(v = KOTLIN_VERSION),
+    ],
+    "sha256": KOTLINC_RELEASE_SHA,
+}
+
+kotlin_repositories(compiler_release = KOTLINC_RELEASE)
+```
+
+## Third party dependencies 
+_(e.g. maven artifacts)_
+
+Third party (external) artifacts can be brought in with systems such as [rules_jvm_external](https://github.com/bazelbuild/rules_jvm_external) or [bazel_maven_repository](https://github.com/square/bazel_maven_repository) or [bazel-deps](https://github.com/johnynek/bazel-deps), but make sure the version you use doesn't naively use java_import, as this will cause bazel to make an interface-only (ijar), or ABI jar, and the native ijar tool does not know about kotlin metadata with respect to inlined functions, and will remove method bodies inappropriately.  Recent versions of [rules_jvm_external] and [bazel_maven_repository] are known to work with kotlin.
+
+
+# Bazel Kotlin Rules compatibility
+
+Which version of *rules_kotlin* can you use with which version of Bazel (best
+effort testing)?
+
+| Version | HEAD   |  |
```

Comment:
Yeah - I can remove this since bazel's CI already has a matrix. I didn't have this when I was running CI on my own hardware.  (hard life, not being a googler anymore - you have to pay for your own datacenter usage). :)